### PR TITLE
refactor(SocketIoDriver): Use Rx.Observable.fromEvent()

### PR DESCRIPTION
**Pull request by saneyuki**
This change is [Reviewable](https://reviewable.io/reviews/karen-irc/karen/635)

### CODE REVIEW

**Comment by saneyuki** [comment]
@dokidokivisual r+

**Comment by dokidokivisual** [comment]
:pushpin: Commit 1ac60fb has been approved by `saneyuki`

**Comment by dokidokivisual** [comment]
:hourglass: Testing commit 1ac60fb with merge cf084f6...

**Comment by saneyuki** [comment]
@dokidokivisual r-

File: src/client/script/adapter/SocketIoDriver.ts [hunk 1]
```diff
@@ -28,214 +28,70 @@ import * as Rx from 'rxjs';
...
-            const callback = (e: any) => {
-                observer.error(e);
-            };
-            this._socket.on(topic, callback);
-
-            return new Rx.Subscription(() => {
-                this._socket.off(topic, callback);
-            });
-        });
+        return Rx.Observable.fromEvent(this._socket, 'error');
```
Comment (saneyuki) [inline src/client/script/adapter/SocketIoDriver.ts]: fix

**Comment by dokidokivisual** [comment]
:sunny: Test successful - [travis](https://travis-ci.org/karen-irc/karen/builds/127179525)

**Comment by saneyuki** [comment]
@dokidokivisual r+

**Comment by dokidokivisual** [comment]
:pushpin: Commit f5202ba has been approved by `saneyuki`

**Comment by dokidokivisual** [comment]
:hourglass: Testing commit f5202ba with merge 031000c...

**Comment by dokidokivisual** [comment]
:sunny: Test successful - [travis](https://travis-ci.org/karen-irc/karen/builds/127180134)

**Pull request merged**